keps/sig-api-machinery/3352-aggregated-discovery/README.md

```diff
@@ -282,102 +319,157 @@ not always required) or even code snippets. If there's any ambiguity
 about HOW your proposal will be implemented, this is the place to
 discuss them. -->
 
-We will expose a endpoint `/discovery` that will support JSON, and
-protobuf and gzip compression. The endpoint will serve the aggregated
-discovery document for all types that a Kubernetes cluster supports.
+The current discovery endpoints `/api` and `/apis` will accept a new
+content negotiation type `Accept:
+application/json;as=APIGroupDiscoveryList;v=v1beta1;g=meta.k8s.io` for
+JSON and `Accept:
+application/vnd.kubernetes.protobuf;as=APIGroupDiscoveryList;v=v1beta1;g=meta.k8s.io`
+for protocol buffers. This type represents an aggregated discovery
+document with the `APIGroupDiscoveryList` type.
+
+When the feature graduates to GA, the accept type's version will
+change to `v1` (Eg: `Accept:
+application/json;as=APIGroupDiscoveryList;v=v1;g=meta.k8s.io`)
```

Comment:
What happens if it's talking to a cluster that only knows about v1beta1?

Reply:
Accept headers typically indicate *all* the acceptable responses, in preferred order, so you'd do:

`Accept:
application/json;as=APIGroupDiscoveryList;v=v1;g=meta.k8s.io,application/json;as=APIGroupDiscoveryList;v=v1beta1;g=meta.k8s.io,application/json`, and the server would return the most-preferred content type it was capable of

Reply:
Thank you, I'm still hoping we can have a good behavior to say "I only support the new way" (especially for future clients).

Reply:
for example, with a current API server:

curl -k -v -s https://localhost:6443/apis -H "Accept: application/json;as=APIGroupDiscoveryList;v=v1;g=meta.k8s.io,application/json;as=APIGroupDiscoveryList;v=v1beta1;g=meta.k8s.io,application/json" | jq . | head
...
> GET /apis HTTP/2
> Host: localhost:6443
> user-agent: curl/7.79.1
> accept: application/json;as=APIGroupDiscoveryList;v=v1;g=meta.k8s.io,application/json;as=APIGroupDiscoveryList;v=v1beta1;g=meta.k8s.io,application/json
> 
...
< HTTP/2 200 
< content-type: application/json
< date: Tue, 04 Oct 2022 20:57:24 GMT
< 
...
{
  "kind": "APIGroupList",
  "apiVersion": "v1",
  "groups": [
    {
      "name": "apiregistration.k8s.io",
      "versions": [
        {
          "groupVersion": "apiregistration.k8s.io/v1",
          "version": "v1"
...
```

Reply:
So I guess `application/json` means the old way?

Reply:
And now that I said that, I can see that you have a dangling `application/json` at the end of your request, good job :-)

Reply:
Accept headers are not the most readable...